packages/astro/src/core/ssr/index.ts

```diff
@@ -228,7 +256,7 @@ export async function render(renderers: Renderer[], mod: ComponentInstance, ssrO
       // HACK: inject the direct contents of our `astro/runtime/client/hmr.js` to ensure
       // `import.meta.hot` is properly handled by Vite
       children: await getHmrScript(),
-      injectTo: 'head',
+      injectTo: 'head-prepend',
```

Comment:
Unrelated change: Vite’s HMR gets injected at the very beginning of `<head>` so I figured we’d do the same.